feat(js/angualar): Update Angular docs for v8 (#9908)
getsentry/sentry-docs: docs/platforms/javascript/guides/angular/features/component-tracking.mdx

description: "Learn how Sentry's Angular SDK allows you to monitor the rendering performance of your application and its components."
---

> Sentry's Angular SDK offers a feature to monitor the performance of your Angular components: Component Tracking. Enabling this feature provides you with spans in your transactions that show the initialization and update cycles of your Angular components. This allows you to get a drilled-down view into how your components are behaving so you can identify slow initializations or frequent updates, which might have an impact on your app's performance.

Reviewer: ```suggestion
Sentry's Angular SDK offers a feature to monitor the performance of your Angular components: Component Tracking. Enabling this feature provides you with spans in your transactions that show the initialization and update cycles of your Angular components. This allows you to drill down into how your components are behaving so you can identify slow initializations or frequent updates, which might have an impact on your app's performance.
```

To track your components as part of your transactions, use any (or a combination) of the following options.

> ## Using the `trace` directive

Reviewer 2: Should we not let this be 
```suggestion
## Using `TraceDirective`
```

for consistency, since below we also use `Using TraceClass` and `Using  TraceMethod` ?

Author: My train of thought was that users would "use" the directive mostly by specifying `trace='myComponent'`. The only time they'd actually need `TraceDirective` (the class) is when registering/importing it in their module. So there's a small difference to the decorators semantically. But I don't have strong feelings; we can also go back to the old heading

Reviewer 2: ah I see, makes sense then - I don't know about angular wording, all good then!

Then, in your component's template, add the directive to all components you want to track. Remember to give the `trace` attribute a name, which will be shown in the span's description:

```html {filename:app.component.(ts|html)}
<app-header trace="header"></app-header>
<articles-list trace="articles-list"></articles-list>
<app-footer trace="footer"></app-footer>
```

## Using `TraceClass`

The `TraceClass` decorator tracks the duration between the `OnInit` and `AfterViewInit` lifecycle hooks in components. It adds spans called **`ui.angular.init`** to the currently active transaction. In contrast to the [`trace` directive](#using-the-trace-directive), `TraceClassDecorator` tracks all instances of the component(s) you add it to, creating spans for each component instance.

Just add `TraceClass` decorator with a component name to the components you want to track:

```typescript {filename:header.component.ts} {2,8}
import { Component } from "@angular/core";
import * as Sentry from "@sentry/angular";

@Component({
  selector: "app-header",
  templateUrl: "./header.component.html",
})
@Sentry.TraceClass({ name: "Header" })
export class HeaderComponent {
  // ...
}
```

> Note: Due to code minification in production builds, you need to pass a `name` property to the `TraceClass` decorator. Otherwise, spans will refer to the component as `<unnamed>`

Reviewer: ```suggestion
Note: Due to code minification in production builds, you need to pass a `name` property to the `TraceClass` decorator. Otherwise, spans will refer to the component as `<unnamed>`.
```

  templateUrl: "./login.component.html",
})
export class LoginComponent implements OnChanges {
  @Sentry.TraceMethod({ name: "Login.ngOnChanges" })
  ngOnChanges(changes: SimpleChanges) {
    // ...
  }
}
```

> Note: Due to code minification in production builds, you need to pass a `name` property to the `TraceMethod` decorator. Otherwise, created spans will refer to the component as `<unnamed>`

Reviewer: ```suggestion
Note: Due to code minification in production builds, you need to pass a `name` property to the `TraceMethod` decorator. Otherwise, created spans will refer to the component as `<unnamed>`.
```
